pkg/ir/spec.go

```diff
@@ -16,39 +16,50 @@ import (
 )
 
 type (
-	ID          = string
-	NamedEntity string
+	ID           = string
+	NamedEntity  string
+	ResourceType string
 
 	Spec struct {
 		// Required connections
-		Connections []Connection
+		Connections []*Connection
 
-		Defs Definitions
+		Defs *Definitions
+
+		*BlockedResources
 	}
 
 	Connection struct {
 		// Egress resource
-		Src Resource
+		Src *LocalRemotePair
 
 		// Ingress resource
-		Dst Resource
+		Dst *LocalRemotePair
 
 		// Allowed protocols
-		TrackedProtocols []TrackedProtocol
+		TrackedProtocols []*TrackedProtocol
 
 		// Provenance information
 		Origin fmt.Stringer
 	}
 
-	Resource struct {
+	// LocalRemotePair holds a local resource and the remote CIDRs it should be connected to
+	LocalRemotePair struct {
 		// Symbolic name of resource, if available
-		Name string
+		Name *string
+
+		LocalCidrs []*NamedAddrs
 
-		// list of CIDR / Ip addresses.
-		IPAddrs []*netset.IPBlock
+		// Cidr list
+		RemoteCidrs []*NamedAddrs
```

Comment:
I'm still confused about the members `LocalCidrs` and `RemoteCidrs` and what's the difference between them.
I'm now pretty sure though, "Local" and "Remote" are not the right terms to use. 
Maybe `SpecEndpoints` and `ProcessedEndpoints`?
We need more discussion here.

Reply:
I don't think `SpecEndpoints` and `ProcessedEndpoints` are good names